src/DistributedComputations/transposable_field.jl

```diff
@@ -0,0 +1,166 @@
+                          (; yz = yzcounts, xy = xycounts),
+                          (; yz = yzcomm,   xy = xycomm))
+end
+
+#####
+##### Twin transposed grid
+#####
+
+"""
+    twin_grid(grid::DistributedGrid; free_dimension = :y)
+
+Constructs a "twin" grid based on the provided distributed `grid` object.
+The twin grid is a grid that discretizes the same domain of the original grid, just with a different partitioning strategy 
+whereas the "free dimension" (i.e. the  non-partitioned dimension) is specified by the keyword argument `free_dimension`.
+This could be either `:x` or `:y`.
+
+Note that `free_dimension = :z` will return the original grid as we do not allow partitioning in the `z` direction 
+"""
+function twin_grid(grid::DistributedGrid; free_dimension = :y)
+
+    arch = grid.architecture
+    ri, rj, rk = arch.local_index
+
+    R = arch.ranks
+
+    nx, ny, nz = n = size(grid)
+    Nx, Ny, Nz = global_size(arch, n)
+
+    TX, TY, TZ = topology(grid)
+
+    TX = reconstruct_global_topology(TX, R[1], ri, rj, rk, arch.communicator)
+    TY = reconstruct_global_topology(TY, R[2], rj, ri, rk, arch.communicator)
+    TZ = reconstruct_global_topology(TZ, R[3], rk, ri, rj, arch.communicator)
+
+    x = cpu_face_constructor_x(grid)
+    y = cpu_face_constructor_y(grid)
+    z = cpu_face_constructor_z(grid)
+
+    ## This will not work with 3D parallelizations!!
+    xG = R[1] == 1 ? x : assemble_coordinate(x, nx, R[1], ri, rj, rk, arch.communicator)
+    yG = R[2] == 1 ? y : assemble_coordinate(y, ny, R[2], rj, ri, rk, arch.communicator)
+    zG = R[3] == 1 ? z : assemble_coordinate(z, nz, R[3], rk, ri, rj, arch.communicator)
+
+    child_arch = child_architecture(arch)
+
+    FT = eltype(grid)
+
+    if free_dimension == :y
+        ranks = R[1], 1, R[2]
+
+        nnx, nny, nnz = nx, Ny, nz ÷ ranks[3]
+
+        if (nnz * ranks[3] < Nz) && (rj == ranks[3])
+            nnz = Nz - nnz * (ranks[3] - 1)
+        end
+    elseif free_dimension == :x
+        ranks = 1, R[1], R[2]
+
+        nnx, nny, nnz = Nx, Ny ÷ ranks[2], nz ÷ ranks[3]
+
+        if (nny * ranks[2] < Ny) && (ri == ranks[2])
+            nny = Ny - nny * (ranks[2] - 1)
+        end
+    elseif free_dimension == :z
+        #TODO: a warning here?
+        return grid
+    end
+
+    new_arch  = Distributed(child_arch; partition = Partition(ranks...))
+    global_sz = global_size(new_arch, (nnx, nny, nnz))
+
+    return construct_grid(grid, new_arch, FT; 
+                          size = global_sz, 
+                        extent = (xG, yG, zG),
+                      topology = (TX, TY, TZ))
+end
+
+construct_grid(::RectilinearGrid, arch, FT; size, extent, topology) = 
+        RectilinearGrid(arch, FT; size, 
+                        x = extent[1],
+                        y = extent[2], 
+                        z = extent[3],
+                        topology)
+
+construct_grid(::LatitudeLongitudeGrid, arch, FT; size, extent, topology) = 
```

Comment:
confused about this in this PR. Do we support transforms on lat lon grids?

Reply:
we do not support transforms, but we do support transposes.

Reply:
Maybe write this in a comment